python/ray/tests/test_reference_counting.py

```diff
@@ -16,13 +16,39 @@
 logger = logging.getLogger(__name__)
 
 
+@pytest.fixture
+def one_worker_100MiB(request):
+    yield ray.init(num_cpus=1, object_store_memory=100 * 1024 * 1024)
+    ray.shutdown()
+
+
+def _fill_object_store_and_get(oid, succeed=True, object_MiB=40,
+                               num_objects=5):
+    for _ in range(num_objects):
+        ray.put(np.zeros(object_MiB * 1024 * 1024, dtype=np.uint8))
+
+    if type(oid) is bytes:
+        oid = ray.ObjectID(oid)
+
+    if succeed:
+        ray.get(oid)
+    else:
+        if oid.is_direct_call_type():
+            with pytest.raises(ray.exceptions.RayTimeoutError):
+                ray.get(oid, timeout=0.1)
+        else:
+            with pytest.raises(ray.exceptions.UnreconstructableError):
+                ray.get(oid)
+
+
 def _check_refcounts(expected):
     actual = ray.worker.global_worker.core_worker.get_all_reference_counts()
     assert len(expected) == len(actual)
     for object_id, (local, submitted) in expected.items():
-        assert object_id in actual
-        assert local == actual[object_id]["local"]
-        assert submitted == actual[object_id]["submitted"]
+        hex_id = object_id.hex().encode("ascii")
```

Comment:
This was a change that I made to prevent the ObjectID from entering scope in python. We can merge it with this PR, but you'll need to add [this change](https://github.com/ray-project/ray/blob/29b85ba8eab1fb933b35d696d116048d3080baf6/python/ray/_raylet.pyx#L1072) also.